In [1]:
import pandas as pd

import plotly.express as pe

In [2]:
link = "https://raw.githubusercontent.com/techficent/PythonDABatch/main/titanic.csv"
df = pd.read_csv(link)
df

Unnamed: 0  PassengerId  Survived  Pclass  \
0             0            1         0       3   
1             1            2         1       1   
2             2            3         1       3   
3             3            4         1       1   
4             4            5         0       3   
..          ...          ...       ...     ...   
886         886          887         0       2   
887         887          888         1       1   
888         888          889         0       3   
889         889          890         1       1   
890         890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 13 columns]

### Printing some columns

In [3]:
df[          ["Name", "Pclass", "Age", "Fare"]             ] 

Name  Pclass   Age     Fare
0                              Braund, Mr. Owen Harris       3  22.0   7.2500
1    Cumings, Mrs. John Bradley (Florence Briggs Th...       1  38.0  71.2833
2                               Heikkinen, Miss. Laina       3  26.0   7.9250
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)       1  35.0  53.1000
4                             Allen, Mr. William Henry       3  35.0   8.0500
..                                                 ...     ...   ...      ...
886                              Montvila, Rev. Juozas       2  27.0  13.0000
887                       Graham, Miss. Margaret Edith       1  19.0  30.0000
888           Johnston, Miss. Catherine Helen "Carrie"       3   NaN  23.4500
889                              Behr, Mr. Karl Howell       1  26.0  30.0000
890                                Dooley, Mr. Patrick       3  32.0   7.7500

[891 rows x 4 columns]

In [4]:
df[           ["Name","Fare"]                ]

Name     Fare
0                              Braund, Mr. Owen Harris   7.2500
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  71.2833
2                               Heikkinen, Miss. Laina   7.9250
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  53.1000
4                             Allen, Mr. William Henry   8.0500
..                                                 ...      ...
886                              Montvila, Rev. Juozas  13.0000
887                       Graham, Miss. Margaret Edith  30.0000
888           Johnston, Miss. Catherine Helen "Carrie"  23.4500
889                              Behr, Mr. Karl Howell  30.0000
890                                Dooley, Mr. Patrick   7.7500

[891 rows x 2 columns]

### shape of data

In [5]:
df.shape

(891, 13)

### list of columns

In [8]:
df.columns

Index(['Unnamed: 0', 'PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age',
       'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

### information of data storage

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 13 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Unnamed: 0   891 non-null    int64  
 1   PassengerId  891 non-null    int64  
 2   Survived     891 non-null    int64  
 3   Pclass       891 non-null    int64  
 4   Name         891 non-null    object 
 5   Sex          891 non-null    object 
 6   Age          714 non-null    float64
 7   SibSp        891 non-null    int64  
 8   Parch        891 non-null    int64  
 9   Ticket       891 non-null    object 
 10  Fare         891 non-null    float64
 11  Cabin        204 non-null    object 
 12  Embarked     889 non-null    object 
dtypes: float64(2), int64(6), object(5)
memory usage: 90.6+ KB


### counting missing values

In [11]:
df.isna().sum()

Unnamed: 0       0
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

### CATEGORICAL V/S REAL-VALUE

price of a car:   real-valued columns

                    ----> fixed number of options ARE NOT POSSIBLE
                    ----> minimum and maximum ARE POSSIBLE

month in a calendar year  : categorical
                            ----> select number of options are valid (fixed options)
                            -----> min & max cannot be calculated


In [13]:
df[   ['Pclass']   ] #categorical data!

Pclass
0         3
1         1
2         3
3         1
4         3
..      ...
886       2
887       1
888       3
889       1
890       3

[891 rows x 1 columns]

In [12]:
df[         ['Sex', "Fare"]             ]
#gender is category and Fare is real-valued

Sex     Fare
0      male   7.2500
1    female  71.2833
2    female   7.9250
3    female  53.1000
4      male   8.0500
..      ...      ...
886    male  13.0000
887  female  30.0000
888  female  23.4500
889    male  30.0000
890    male   7.7500

[891 rows x 2 columns]

#### number of unique values in a column

In [14]:
df.nunique()

Unnamed: 0     891
PassengerId    891
Survived         2
Pclass           3
Name           891
Sex              2
Age             88
SibSp            7
Parch            7
Ticket         681
Fare           248
Cabin          147
Embarked         3
dtype: int64

# real-value columns statistics
            ----> average
            ---->min
            ---->max
            ---->percentile

In [ ]:
Maximum possible marks (100 marks)



total students --- 1000


actual maximum value :     90  (highest marks obtained)




harshit 's marks 81
percentage:?
percentile:?



actual  minimum value :         20 (lowest marks obtained)

### percentage: out of MAXIMUM POSSIBLE VALUE


harshit's percentage:   81 * 100
                      ---------                      = 81 percentage!!! (harshit's independent capability)
                       total value (100)


## percentile: out of maximum observed value

harshit's percentile:    81 * 100
                     ----------------                       = 90 percentile (harshit's value compared to other data point)
                       max observable value : 90

### count: actual valid data points in the column
### mean : average of the column
### std:  standard deviation
### min:  minimum observed value in the column
### max: maxium observed value in the column
### 25% : 25th percentile of the data

In [17]:
df[          ['Age', 'Fare']           ].describe()

Age        Fare
count  714.000000  891.000000
mean    29.699118   32.204208
std     14.526497   49.693429
min      0.420000    0.000000
25%     20.125000    7.910400
50%     28.000000   14.454200
75%     38.000000   31.000000
max     80.000000  512.329200

## age column has 25th percentile at 20.125

if I observe the data for ALL PASSENGERS, 1/4th of my passengers's age will be below or equal to 20.125


## age column has 50th percentile at 28.0000
if I observed data for ALL PASSENGERS, half (1/2) of my passenger's age will be below or
equal to 28.000